### PR TITLE
feat: implement running experiments on setTimeout

### PR DESCRIPTION
**Pull request by GuyAb**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                      |
| Breaking change |✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                          |

Related issues: #634 
Further  information:

### CODE REVIEW

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        logger.info('kubernetes token from env var was not provided. will use: ' + TOKEN_PATH);
+        const token = fs.readFileSync(TOKEN_PATH);
+        headers.Authorization = 'bearer ' + token.toString();
+    } catch (error) {
+        logger.warn(error, 'Failed to get kubernetes token from: ' + TOKEN_PATH);
+    }
+}
+
+module.exports.runChaosExperiment = async (kubernetesExperimentConfig) => {
+    const resourceTypeName = kubernetesExperimentConfig.kind.toLowerCase();
```
Comment (kerenfi) [inline src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js:26]: resourceTypeName => resourceKindName

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const url = util.format('%s/apis/chaos-mesh.org/v1/namespaces/%s/%s', kubernetesUrl, kubernetesNamespace, resourceTypeName);
```
Comment (kerenfi) [inline src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js:27]: maybe good to add logs of before and after running

File: src/jobs/models/jobExperimentsHandler.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 
 const chaosExperimentsDbConnector = require('../../chaos-experiments/models/database/databaseConnector'),
```
Comment (kerenfi) [inline src/jobs/models/jobExperimentsHandler.js:2]: probalby redundent as we use manager

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js (hunk 1 above)
Comment (GuyAb) [reply src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js]: fixed

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js (hunk 2 above)
Comment (GuyAb) [reply src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js]: already exists

File: tests/unit-tests/chaos-experiments/models/chaosExperimentsManager-test.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ const rewire = require('rewire');
 
 const database = require('../../../../src/chaos-experiments/models/database/databaseConnector');
 const manager = rewire('../../../../src/chaos-experiments/models/chaosExperimentsManager');
```
Comment (kerenfi) [inline tests/unit-tests/chaos-experiments/models/chaosExperimentsManager-test.js:8]: fyi - we don't use rewire functionality here - (__set__ )
I prefer to remove it as this is unhealthy methodology in testing

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+const util = require('util');
+const fs = require('fs');
+const kubernetesConfig = require('../../../config/kubernetesConfig');
+const logger = require('../../../common/logger');
+const requestSender = require('../../../common/requestSender');
+const kubernetesUrl = kubernetesConfig.kubernetesUrl;
+const kubernetesNamespace = 'apps'; // kubernetesConfig.kubernetesNamespace;
```
Comment (kerenfi) [inline src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js]: fix to not be hardcoded but take it from the kubeObject

File: tests/unit-tests/chaos-experiments/models/kubernetes/chaosExperimentConnector-test.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                    action: 'http-kill',
+                    duration: '1m'
+                }
+            }
+        };
+        requestSenderSendStub.resolves({ metadata: { name: 'firstHttpChaos', uid: 'some_uuid' }, namespace: 'default' });
+        await chaosExperimentConnector.runChaosExperiment(kubernetesJobConfig);
+        requestSenderSendStub.callCount.should.eql(1);
+        requestSenderSendStub.args[0][0].should.eql({
+            url: 'localhost:80/apis/chaos-mesh.org/v1alpha1/namespaces/default/httpchaos',
```
Comment (kerenfi) [inline tests/unit-tests/chaos-experiments/models/kubernetes/chaosExperimentConnector-test.js:80]: test default namespace?

File: src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js (hunk 5 above)
Comment (GuyAb) [reply src/chaos-experiments/models/kubernetes/chaosExperimentConnector.js]: fixed

File: tests/unit-tests/chaos-experiments/models/kubernetes/chaosExperimentConnector-test.js (hunk 6 above)
Comment (GuyAb) [reply tests/unit-tests/chaos-experiments/models/kubernetes/chaosExperimentConnector-test.js]: changed to apps

**Pull request merged**